dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/version/VersionPopup.java

```diff
@@ -0,0 +1,99 @@
+ * ******************************************************************************
+ */
+
+package com.spectralogic.dsbrowser.gui.components.version;
+
+import com.spectralogic.ds3client.Ds3Client;
+import com.spectralogic.ds3client.commands.GetBucketRequest;
+import com.spectralogic.ds3client.models.Contents;
+import com.spectralogic.dsbrowser.api.services.logging.LogType;
+import com.spectralogic.dsbrowser.api.services.logging.LoggingService;
+import com.spectralogic.dsbrowser.gui.components.ds3panel.Ds3Common;
+import com.spectralogic.dsbrowser.gui.components.ds3panel.ds3treetable.Ds3TreeTableValue;
+import com.spectralogic.dsbrowser.gui.util.AlertService;
+import com.spectralogic.dsbrowser.gui.util.StringConstants;
+import com.spectralogic.dsbrowser.util.GuavaCollectors;
+import javafx.scene.Scene;
+import javafx.scene.image.Image;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.ResourceBundle;
+
+public class VersionPopup {
+    private static final Logger LOG = LoggerFactory.getLogger(VersionPopup.class);
+
+    private final ResourceBundle resourceBundle;
+    private final Ds3Common ds3Common;
+    private final LoggingService loggingService;
+    private final AlertService alertService;
+
+    @Inject
+    public VersionPopup(
+            final ResourceBundle resourceBundle,
+            final Ds3Common ds3Common,
+            final LoggingService loggingService,
+            final AlertService alertService
+    ) {
+       this.resourceBundle = resourceBundle;
+       this.ds3Common = ds3Common;
+       this.loggingService = loggingService;
+       this.alertService = alertService;
+    }
+
+    public void show(final Ds3TreeTableValue item) {
+        final List<VersionItem> versions = getVersioned(item)
+                .stream()
+                .map(contents -> new VersionItem(contents.getKey(), contents.getLastModified(), contents.getVersionId(), contents.getSize()))
+                .collect(GuavaCollectors.immutableList());
+
+        if (versions.isEmpty()) {
+            alertService.warning("unableToGetVersions");
+        } else {
+            final Stage popup = new Stage();
+            final VersionView view = new VersionView(new VersionModel(item.getBucketName(), versions, popup));
+            final Scene popupScene = new Scene(view.getView());
+
+            popup.initModality(Modality.APPLICATION_MODAL);
+            popup.getIcons().add(new Image(StringConstants.DSB_ICON_PATH));
+            popup.setScene(popupScene);
+            popup.setTitle(resourceBundle.getString("versionView"));
+            popup.setAlwaysOnTop(false);
+            popup.setResizable(true);
+            popup.showAndWait();
+        }
+    }
+
+    private List<Contents> getVersioned(final Ds3TreeTableValue item) {
```

Comment:
What thread is this call being made in?  The UI thread or a background thread?

Reply:
Also, is this a call that should be made in the presenter before the popup view is created?

Reply:
My thought was that we shouldn't bother drawing the presenter if this call doesn't work

Reply:
What about the thread that call is made on?  Will it block the UI thread while the call is made?

Reply:
I'll move it out of the UI thread